report/data_sources/hubspot_client.py

```diff
@@ -22,6 +23,21 @@ def __post_init__(self):
             self.key = self.hs_field
 
 
+def chunk(items: Iterable, chunk_size: int):
+    """Return items in multiple chunks of a maximum size.
+
+    :param items: Items to break into chunks
+    :param chunk_size: The maximum size of a chunk (last may be shorter)
+    """
+
+    # Create an iterable we can consume from
+    iter_items = iter(items)
+    # Use the two-argument form of iter to specify when to stop. This will
+    # cause iter to call the lambda until it returns an empty tuple, which
+    # should consume the iterable above.
+    return iter(lambda: tuple(islice(iter_items, chunk_size)), ())
```

Comment:
Is this is code "inspired" by a stack overflow post:

https://stackoverflow.com/questions/312443/how-do-i-split-a-list-into-equally-sized-chunks/312464#312464

Which is in turn taken from the itertools example `def batched` from https://docs.python.org/3/library/itertools.html